sources/models/build.rs

```diff
@@ -4,62 +4,65 @@
 //
 // See README.md to understand the symlink setup.
 
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
 use std::env;
 use std::fs::{self, File};
 use std::io::{self, Write};
 use std::os::unix::fs::symlink;
 use std::path::{Path, PathBuf};
 use std::process;
 
-fn symlink_force<P1, P2>(target: P1, link: P2) -> io::Result<()>
-where
-    P1: AsRef<Path>,
-    P2: AsRef<Path>,
-{
-    // Remove link if it already exists
-    if let Err(e) = fs::remove_file(&link) {
-        if e.kind() != io::ErrorKind::NotFound {
-            return Err(e);
-        }
+const VARIANT_LINK: &str = "src/variant/current";
+const MOD_LINK: &str = "src/variant/mod.rs";
+const VARIANT_ENV: &str = "VARIANT";
+
+fn main() {
+    // Tell cargo when we have to rerun, regardless of early-exit below.
+    println!("cargo:rerun-if-env-changed={}", VARIANT_ENV);
+    println!("cargo:rerun-if-changed={}", VARIANT_LINK);
+    println!("cargo:rerun-if-changed={}", MOD_LINK);
+
+    // This build.rs runs once as a build-dependency of storewolf, and again as a (regular)
+    // dependency of storewolf.  There's no reason to do this work twice.
+    if env::var("CARGO_CFG_TARGET_VENDOR").unwrap_or_else(|_| String::new()) == "bottlerocket" {
+        println!("cargo:warning=Already ran model build.rs for host, skipping for target");
+        process::exit(0);
     }
-    // Link to requested target
-    symlink(&target, &link)
+
+    generate_readme();
+    link_current_variant();
 }
 
 fn link_current_variant() {
     // The VARIANT variable is originally BUILDSYS_VARIANT, set in the top-level Makefile.toml,
     // and is passed through as VARIANT by the top-level Dockerfile.  It represents which OS
     // variant we're building, and therefore which API model to use.
-    let var = "VARIANT";
-    println!("cargo:rerun-if-env-changed={}", var);
-    let variant = env::var(var).unwrap_or_else(|_| {
-        eprintln!("For local builds, you must set the {} environment variable so we know which API model to build against.  Valid values are the directories in variants/, for example \"aws-k8s-1.17\".", var);
+    let variant = env::var(VARIANT_ENV).unwrap_or_else(|_| {
+        eprintln!("For local builds, you must set the {} environment variable so we know which API model to build against.  Valid values are the directories in variants/, for example \"aws-k8s-1.17\".", VARIANT_ENV);
         process::exit(1);
     });
 
     // Point to the source for the requested variant
```

Comment:
looks like this comment was for ` let variant_link = "src/variant/current";` which was existing before and now moved to the top as const.

Reply:
I think this comment is still correct; the next line is setting the target of the link, and this comment is about the target of the link, i.e. where it's pointing to.